src/shared/internal-calendar/internal-week-selection-calendar-day.style.tsx

```diff
@@ -0,0 +1,340 @@
+export interface StyleProps {
+    $disabledDisplay?: boolean;
+    $interactive?: boolean;
+    $overlap?: boolean;
+    $hovered?: boolean;
+    $selected?: boolean;
+}
+
+export interface DayLabelStyleProps extends StyleProps {
+    $variant: DayVariant;
+}
+
+export interface OverflowCircleProps extends Partial<OverflowDisplayProps> {}
+
+interface OverflowDisplayProps extends StyleProps {
+    $position: "left" | "right";
+}
+
+interface InteractiveCircleProps
+    extends DayLabelStyleProps,
+        OverflowCircleProps {}
+
+// =============================================================================
+// STYLING
+// =============================================================================
+export const Wrapper = styled.div`
+    width: 100%;
+    display: grid;
+    grid-template-columns: repeat(7, 1fr);
+    row-gap: 0.25rem;
+`;
+
+export const HeaderCell = styled.div`
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    height: 2.5rem;
+    pointer-events: none;
+    user-select: none;
+`;
+
+export const RowDayCell = styled.div`
+    grid-column: 1 / -1;
+    display: flex;
+`;
+
+export const GrowDayCell = styled.div`
+    display: flex;
+    position: relative;
+    height: 2.5rem;
+    align-items: center;
+    justify-content: center;
+    flex: 1;
+`;
+
+export const OverflowDisplay = styled.div<OverflowDisplayProps>`
+    position: absolute;
+    width: 50%;
+    height: 100%;
+
+    ${(props) => {
+        switch (props.$position) {
+            case "left":
+                return css`
+                    left: 0;
+                `;
+            case "right":
+                return css`
+                    right: 0;
+                `;
+        }
+    }}
+
+    ${(props) => {
+        const { $hovered, $selected } = props;
+
+        if ($hovered) {
+            return css`
+                border-top: 1px dashed ${Color.Accent.Light[4]};
+                border-bottom: 1px dashed ${Color.Accent.Light[4]};
+                background-color: ${Color.Accent.Light[6]};
+            `;
+        }
+
+        if ($selected) {
+            return css`
+                border-top: 1px solid ${Color.Primary};
+                border-bottom: 1px solid ${Color.Primary};
+                background-color: ${Color.Accent.Light[5]};
+            `;
+        }
+    }}
+
+
+    ${(props) => {
+        if (props.$overlap) {
+            return css`
+                border-top: 1px solid ${Color.Primary};
+                border-bottom: 1px solid ${Color.Primary};
+                background-color: ${Color.Neutral[8]};
+                box-shadow: 4px 0px 4px 0px ${Color.Shadow.Accent};
+            `;
+        }
+    }}
+`;
+
+export const OverflowCircle = styled.div<OverflowCircleProps>`
+    display: none;
+    position: absolute;
+    pointer-events: none;
+    background-color: transparent;
+    width: 2.5rem;
+    height: 2.5rem;
+    border-radius: 50%;
+
+    ::before,
+    ::after {
```

Comment:
doesn't seem like `::after` is actually used, can it be removed?

Reply:
ya!